### PR TITLE
Model Creation & Edit, Banner and Start Mapping Layout

### PR DESCRIPTION
**Pull request by jeafreezy**
### What does this PR do ?

This PR implements the model creation flow. The following features are implemented:

1. Model creation and update.
2. Training dataset creation and selection.
4. Training area: Drawing, Upload, Delete, Open in ID Editor, Open in JOSM, Fetching of Labels.
5. Training Settings Configurations.
6. Training request submission.
7. Model enhancement
8. Banner
9. Start mapping layout
10. and many more.

### How to Test

Visit: https://f-a-ir-emmanuels-projects-1886bda9.vercel.app/

### CODE REVIEW

File: frontend/package.json
Comment (omranlm) [inline frontend/package.json:1]: @jeafreezy the amount of dependencies is increasing rapidly, let's just make sure we are using community supported libs

**Review by omranlm — APPROVED** [review]
Thanks @jeafreezy

File: frontend/package.json
Comment (jeafreezy) [reply frontend/package.json:1]: Roger that.

Thanks for your concern @omranlm . The new additions are:

1. **terraformer/wkt**:  We need it to convert the AOI to WKT. (Necessary)
2. **@turf/area**: We need it to calculate the area for the AOI. (Necessary)
3. **@turf/bbox**: We need it to get the bounds for the AOI. (Necessary)
4. **react-dropzone**: We need it for the AOI file's drag-and-drop dialog functionality. (Not really necessary - we can allow the users to just open their file picker when they click on the 'upload' button. However, I must admit that it provides a better UX.)
5. **react-confetti-explosion**: When the model creation is successful, we need it for the confetti explosion. (Not really necessary, too.  However, I must also admit that it provides a good UX).
6. **terra-draw**: For drawing functionality. (Necessary)

**Some suggestions:**

The top three can be delegated to the backend later in the future, therefore eliminating the dependencies.

1. Send the raw AOI GeoJSON feature directly to the backend instead of transforming it to WKT on the client.
2. The backend should calculate the area and return it as part of the GeoJSON feature properties.
3. The backend should return the bbox as properties of the GeoJSON feature.

Rest assured, I will ensure all dependencies are community-supported libs and lightweight.

Comment (kshitijrajsharma) [reply frontend/package.json:1]: >Some suggestions:

The top three can be delegated to the backend later in the future, therefore eliminating the dependencies.

    Send the raw AOI GeoJSON feature directly to the backend instead of transforming it to WKT on the client.
    The backend should calculate the area and return it as part of the GeoJSON feature properties.
    The backend should return the bbox as properties of the GeoJSON feature.
@jeafreezy  Raise this as an issue ! I will resolve when I can

Comment (jeafreezy) [reply frontend/package.json:1]: Done in #304  and #303 .
